app/models/user.rb

```diff
@@ -65,15 +65,15 @@ class User < ApplicationRecord
   attr_accessor :asserted_attributes, :email
 
   def confirmed_email_addresses
-    email_addresses.where.not(confirmed_at: nil).order('last_sign_in_at DESC NULLS LAST')
+    email_addresses.confirmed
```

Comment:
How are we ensuring `last_sign_in_at` ordering with this change?

Reply:
This line by itself does not ensure that. The change to `User#last_sign_in_email_address` on line 524 does it.

Reply:
After further discussion with @aduth, we decided to leave the sort order in, even though the tests pass. We plan to clean this up in a separate PR.